hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2385,6 +2526,42 @@ object NDArrayEmitter {
       ))
     }
   }
+
+  def matmulShape(leftShape: Array[Code[Long]], rightShape: Array[Code[Long]]): (Code[Unit], Array[Code[Long]]) = {
```

Comment:
I'm a wee bit said that we have these two match statements that have the same four cases in both matmul shape and in the index stuff. Is it possible to unify these? A little tricky because we don't have the index vars on hand

Reply:
I'm reluctant to unify them. `matmulShape` will exist across all `matmul` implementations, indexing stuff will not be relevant in a BLAS implementation.

Reply:
I did replace my original implementation with the simpler one you proposed though.